org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/DashboardController.java

```diff
@@ -1643,13 +1649,48 @@ private ReportEngine runReport(int AD_Process_ID, int AD_PrintFormat_ID, String
 	 * @throws Exception
 	 */
 	private ReportData generateReport(int AD_Process_ID, int AD_PrintFormat_ID, String parameters, Component component, String contextPath) throws Exception {
+		MProcess process = MProcess.get(Env.getCtx(), AD_Process_ID);
+		File file = null;
+		if(process.getJasperReport() != null) {
+			file = runJasperReport(process, parameters);
+			return new ReportData(new AMedia(process.getName(), "html", "text/html", file, false), -1);
+		}
+			
 		ReportEngine re = runReport(AD_Process_ID, AD_PrintFormat_ID, parameters);
 		if(re == null)
 			return null;
-		File file = FileUtil.createTempFile(re.getName(), ".html");		
+		file = FileUtil.createTempFile(re.getName(), ".html");		
 		re.createHTML(file, false, AEnv.getLanguage(Env.getCtx()), new HTMLExtension(contextPath, "rp", 
 				component.getUuid(), String.valueOf(AD_Process_ID)));
-		return new ReportData(new AMedia(re.getName(), "html", "text/html", file, false), re);
+		return new ReportData(new AMedia(process.getName(), "html", "text/html", file, false), re.getPrintData() != null ? re.getPrintData().getRowCount(false) : 0);
+	}
+
+	private File runJasperReport(MProcess process, String parameters) {
+		MPInstance pInstance = new MPInstance(Env.getCtx(), process.getAD_Process_ID(), 0, 0, null);
+		pInstance.setIsProcessing(true);
+		pInstance.saveEx();
+		try {
+			if(!fillParameter(pInstance, parameters))
+				return null;
+			//
+			if(process.getJasperReport() == null)
+				throw new RuntimeException("no Jasper Report File provided. Process "+process.getName());
```

Comment:
This type of checking should be first in the method (i.e before the MPInstance pInstance line). However, looking at the flow of the code, isn't this should never happens (you already has the if(process.getJasperReport() != null) checking above).

Reply:
actually, on line 1654, i already check it. so if getJasperReport is null, the the method runJasperReport not execute. should i still add check getJasperReport on first line of runJasperReport method or just remove check getJasperReport on code above?

Reply:
yes, I means the checking here is redundant.